presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaPageSource.java

```diff
@@ -0,0 +1,400 @@
+                ordinalPosition++;
+            }
+        }
+    }
+
+    private void addTablePrivilegesRecordsFor(QualifiedTablePrefix prefix)
+    {
+        List<GrantInfo> grants = ImmutableList.copyOf(listTablePrivileges(session, metadata, accessControl, prefix));
+        for (GrantInfo grant : grants) {
+            addRecord(
+                    grant.getGrantor().map(PrestoPrincipal::getName).orElse(null),
+                    grant.getGrantor().map(principal -> principal.getType().toString()).orElse(null),
+                    grant.getGrantee().getName(),
+                    grant.getGrantee().getType().toString(),
+                    prefix.getCatalogName(),
+                    grant.getSchemaTableName().getSchemaName(),
+                    grant.getSchemaTableName().getTableName(),
+                    grant.getPrivilegeInfo().getPrivilege().name(),
+                    grant.getPrivilegeInfo().isGrantOption() ? "YES" : "NO",
+                    grant.getWithHierarchy().map(withHierarchy -> withHierarchy ? "YES" : "NO").orElse(null));
+        }
+    }
+
+    private void addTablesRecordsFor(QualifiedTablePrefix prefix)
+    {
+        Set<SchemaTableName> tables = listTables(session, metadata, accessControl, prefix);
+        Set<SchemaTableName> views = listViews(session, metadata, accessControl, prefix);
+
+        for (SchemaTableName name : union(tables, views)) {
+            // if table and view names overlap, the view wins
+            String type = views.contains(name) ? "VIEW" : "BASE TABLE";
+            addRecord(
+                    prefix.getCatalogName(),
+                    name.getSchemaName(),
+                    name.getTableName(),
+                    type,
+                    null);
+        }
+    }
+
+    private void addViewsRecordsFor(QualifiedTablePrefix prefix)
+    {
+        for (Map.Entry<QualifiedObjectName, ConnectorViewDefinition> entry : metadata.getViews(session, prefix).entrySet()) {
+            addRecord(
+                    entry.getKey().getCatalogName(),
+                    entry.getKey().getSchemaName(),
+                    entry.getKey().getObjectName(),
+                    entry.getValue().getOriginalSql());
+        }
+    }
+
+    private void addRecord(Object... values)
+    {
+        rawPageBuilder.declarePosition();
+        for (int i = 0; i < columnTypes.size(); i++) {
+            writeNativeValue(columnTypes.get(i), rawPageBuilder.getBlockBuilder(i), values[i]);
+        }
+
+        if (rawPageBuilder.isFull()) {
+            flushPageBuilder();
+        }
+    }
+
+    private void flushPageBuilder()
+    {
+        if (!rawPageBuilder.isEmpty()) {
+            rawPages.add(rawPageBuilder.build());
+            memoryUsageBytes += rawPageBuilder.getRetainedSizeInBytes();
+            rawPageBuilder.reset();
+        }
+    }
+
+    private static InformationSchemaTableType getTableTypeBySchemaTableName(SchemaTableName schemaTableName)
```

Comment:
It think you could move this method to `InformationSchemaTableType`. Like `InformationSchemaTableType.of(schemaTableName)`